exp/controllers/azuremanagedcluster_controller.go

```diff
@@ -45,10 +51,27 @@ type AzureManagedClusterReconciler struct {
 }
 
 func (r *AzureManagedClusterReconciler) SetupWithManager(mgr ctrl.Manager, options controller.Options) error {
-	return ctrl.NewControllerManagedBy(mgr).
+	azManagedCluster := &infrav1exp.AzureManagedCluster{}
+	c, err := ctrl.NewControllerManagedBy(mgr).
 		WithOptions(options).
-		For(&infrav1exp.AzureManagedCluster{}).
-		Complete(r)
+		For(azManagedCluster).
+		WithEventFilter(predicates.ResourceNotPaused(r.Log)). // don't queue reconcile if resource is paused
+		Build(r)
+	if err != nil {
+		return errors.Wrapf(err, "error creating controller")
+	}
+
+	// Add a watch on clusterv1.Cluster object for unpause & ready notifications.
```

Comment:
This should just check for unpaused, otherwise we might filter out events that we want to act on, since this controller is responsible for providing the value that populates cluster.status.infrastructureReady
```suggestion
	// Add a watch on clusterv1.Cluster object for unpause notifications.
```

Reply:
Changed to unpaused only.